blockscout-ens/graph-node/subgraphs/zns-subgraph/src/zns-registry.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  MintedDomain as MintedDomainEvent,
+  PrimaryDomainSet as PrimaryDomainSetEvent,
+  RenewedDomain as RenewedDomainEvent,
+  Transfer as TransferEvent,
+} from "../generated/ZNSRegistry/ZNSRegistry";
+import {
+  Domain,
+  Account,
+  Transfer,
+  Registration,
+  ExpiryExtended,
+} from "../generated/schema";
+import { crypto, Bytes } from "@graphprotocol/graph-ts";
+
+export function handleMintedDomain(event: MintedDomainEvent): void {
+  let domain = new Domain(event.params.tokenId.toString());
```

Comment:
ID of domain should be in ERC-137 format: https://github.com/ethereum/ercs/blob/master/ERCS/erc-137.md

but tokenId is just autoincremented integer

Reply:
Ok so we can do that, rather than the token ID, the id for each domain can be the namehash of the domain. Would you like it to be the namehash of "zns.poly" or just "zns"?

Reply:
namehash should be from the whole domain. Note that it's not just hash from string `"zns.poly"`. the algorithm described here: https://github.com/ethereum/ercs/blob/master/ERCS/erc-137.md#namehash-algorithm

for example subgraph for rootstock uses this function: https://github.com/blockscout/blockscout-rs/blob/ea31f1c5e0f88a586252fff9aa6fd29c6252fd80/blockscout-ens/graph-node/subgraphs/rns-subgraph/src/utils.ts#L104-L119